_posts/2019-04-24-istio-service-mesh-s2s.md

```diff
@@ -0,0 +1,431 @@
+This is shown in the following drawing.
+
+<img class="image fit" style="margin:0px auto; max-width: 417px;" src="{{ '/img/2019-04-14-istio-service-mesh-s2s/istio-rbac-refused.png' | prepend: site.baseurl }}" alt="RBAC refuses connection" />
+
+The following manifest creates a role that allows access to the CouchDB service for all GET requests on any given path.
+Note that the full service name is used in the services specification, this is currently required by Istio.
+This is only needed for Istio to identify the traffic, short names can still be used to access the service.
+By applying this service role, nothing will change to the requests to the test-app since the role is not yet connected to the service account of the test-app service. 
+
+```yaml
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRole
+metadata:
+  name: couchdb-role
+  namespace: with-istio
+spec:
+  rules:
+  - services: ["couchdb.with-istio.svc.cluster.local"]
+    methods: ["GET"]
+    paths: ['*']
+```
+
+So next, we link the new role to the service account of the test-app service.
+This is done through a service role binding.
+There are two sections to this binding: the role and the subjects.
+The role is the one that was created using the previous manifest. 
+The subjects can be any identity known to Istio. 
+In the demo scenario, only the service accounts are known.
+Istio defines a service account as a user identity. 
+As with the services names, the service account reference needs to be the full reference scoped towards the cluster.
+This allows services from outside of the namespaces to be specified as well.
+
+```yaml
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRoleBinding
+metadata:
+  name: bind-test-app-service-couchdb-role
+  namespace: with-istio
+spec:
+  subjects:
+  - user: "cluster.local/ns/with-istio/sa/test-app-test-app-chart"
+  roleRef:
+    kind: ServiceRole
+    name: "couchdb-role"
+```
+
+After applying the last manifest, requests should again be authorized and allowed to CouchDB.
+
+## Conclusion
+
+This demo showed how Istio can be used to enable strong authentication and authorization for services in a Kubernetes cluster. 
+It does this by enabling mTLS connections between the Envoy sidecar proxies.
+By creating roles, access to services can be granted or blocked. 
+A role can be connected to a service account to allow access. 
+
+In a nutshell, Istio allows cluster admins to enable secure communication, and strong authentication and authorization mechanisms on their Kubernetes cluster without having to manage all kinds of certificates, usernames and passwords. 
+The application developers don't need to adopt their application in order to communicate securly in the cluster, nor do they have to change their deployment configuration to enable the service mesh.
+
+This blogpost only highlighted a portion of the features of Istio. 
+Security is only a part of the featureset. 
```

Comment:
Security is only a part of the *feature set*.

Reply:
Updated